File: src/main/java/com/wishai/xzrtw/controller/InfoController.java
package com.wishai.xzrtw.controller;

import com.wishai.xzrtw.model.Progress;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.*;

@Controller
@RequestMapping("info")
@SessionAttributes("status")
public class InfoController {

    @ModelAttribute("status")
    public Progress status() {
        return new Progress();
    }

    @GetMapping("/upload")
    @ResponseBody
    public String uploadProgressInfo(@ModelAttribute("status") Progress status) {
        if (status.isEmpty()) {
            return "{}";
        }
        return status.toString();
    }
}

File: src/main/java/com/wishai/xzrtw/controller/InfoController.kt
package com.wishai.xzrtw.controller

import com.wishai.xzrtw.model.Progress
import org.springframework.stereotype.Controller
import org.springframework.web.bind.annotation.*

@Controller
@RequestMapping("info")
@SessionAttributes("status")
class InfoController {

    @ModelAttribute("status")
    fun status(): Progress {
        return Progress()
    }

    @GetMapping("/upload")
    @ResponseBody
    fun uploadProgressInfo(@ModelAttribute("status") status: Progress): String {
        return if (status.isEmpty) {
            "{}"
        } else {
            status.toString()
        }
    }
}
